Tests/Common/Data/Custom/TradingEconomicsTests.cs

```diff
@@ -72,10 +75,13 @@ public void CalendarFilterAppliedCorrectly(string eventName, string expected)
         public void DeserializesProperly()
         {
             var instance = new TradingEconomicsCalendar();
+            var csvBytes = Encoding.UTF8.GetBytes(JsonConvert.DeserializeObject<List<TradingEconomicsCalendar>>(TestCalendarJson).First().ToCsv());
+            var calendarStream = new StreamReader(new MemoryStream(csvBytes));
```

Comment:
I think we should add a _new_ test for the new StreamReader Reader implementation and keep a test for the other existing reader too, maybe even assert the generate the same result